npm/private/npm_package_store.bzl

```diff
@@ -205,25 +205,23 @@ def _npm_package_store_impl(ctx):
         else:
             package_store_directory = ctx.actions.declare_directory(package_store_directory_path)
             if utils.is_tarball_extension(src.extension):
-                # npm packages are always published with one top-level directory inside the tarball, tho the name is not predictable
-                # we can use the --strip-components 1 argument with tar to strip one directory level
-                args = ctx.actions.args()
-                args.add("--extract")
-                args.add("--no-same-owner")
-                args.add("--no-same-permissions")
-                args.add("--strip-components")
-                args.add(str(1))
-                args.add("--file")
-                args.add(src.path)
-                args.add("--directory")
-                args.add(package_store_directory.path)
-
+                # npm packages are always published with one top-level directory inside the tarball,
+                # tho the name is not predictable we can use the --strip-components 1 argument with
+                # tar to strip one directory level. Some packages have directory permissions missing
+                # executable which make the directories not listable (pngjs@5.0.0 for example). Run
+                # `chmod -R a+X` to fix up these packages (https://stackoverflow.com/a/14634721).
+                # See https://github.com/aspect-build/rules_js/issues/1637 for more info.
                 bsdtar = ctx.toolchains["@aspect_bazel_lib//lib:tar_toolchain_type"]
-                ctx.actions.run(
-                    executable = bsdtar.tarinfo.binary,
+                ctx.actions.run_shell(
+                    tools = [bsdtar.tarinfo.binary],
                     inputs = depset(direct = [src], transitive = [bsdtar.default.files]),
                     outputs = [package_store_directory],
-                    arguments = [args],
+                    command = "$1 --extract --no-same-owner --no-same-permissions --strip-components 1 --file $2 --directory $3 && chmod -R a+X $3",
+                    arguments = [
+                        bsdtar.tarinfo.binary.path,
```

Comment:
I think we can drop the `.path` and bazel will expand it when the action actually runs? See https://bazel.build/rules/performance#ctx-actions-args

Reply:
Will try it out